sweep/sweeper.go

```diff
@@ -1163,20 +1163,26 @@ func (s *UtxoSweeper) sweep(inputs inputSet, feeRate chainfee.SatPerKWeight,
 	}
 
 	// Create sweep tx.
-	tx, _, err := createSweepTx(
+	tx, fee, err := createSweepTx(
 		inputs, nil, s.currentOutputScript, uint32(currentHeight),
 		feeRate, s.cfg.MaxFeeRate.FeePerKWeight(), s.cfg.Signer,
 	)
 	if err != nil {
 		return fmt.Errorf("create sweep tx: %v", err)
 	}
 
+	tr := &TxRecord{
+		Txid:    tx.TxHash(),
+		FeeRate: uint64(feeRate),
+		Fee:     uint64(fee),
+	}
+
 	// Add tx before publication, so that we will always know that a spend
 	// by this tx is ours. Otherwise if the publish doesn't return, but did
 	// publish, we loose track of this tx. Even republication on startup
 	// doesn't prevent this, because that call returns a double spend error
 	// then and would also not add the hash to the store.
-	err = s.cfg.Store.StoreTx(tx.TxHash())
+	err = s.cfg.Store.StoreTx(tr)
```

Comment:
use Txid and the TxRecord here as two parameters so that we don't need to use the txid in the TxRecord ?

Reply:
I actually think it's nice we can access it via `TxRecord.Txid`.